contracts/lbp/LBPManager.sol

```diff
@@ -0,0 +1,290 @@
+     * @param _startTimeEndTime         Array containing two parameters:
+                                            - Start time for the LBP.
+                                            - End time for the LBP.
+     * @param _endWeights               Sorted array to match the _tokenList, containing two parametes:
+                                            - The end weight for the project token in the LBP.
+                                            - The end weight for the funding token in the LBP.
+     * @param _swapFeePercentage        Percentage of fee paid for every swap in the LBP.
+     * @param _primeDaoFeePercentage    Percentage of fee paid to PrimeDao for providing the service of the LBP Manager.
+     */
+    function initializeLBP(
+        address _LBPFactory,
+        address _beneficiary,
+        string memory _name,
+        string memory _symbol,
+        IERC20[] memory _tokenList,
+        uint256[] memory _amounts,
+        uint256[] memory _weights,
+        uint256[] memory _startTimeEndTime,
+        uint256[] memory _endWeights,
+        uint256 _swapFeePercentage,
+        uint256 _primeDaoFeePercentage
+    ) external returns (address) {
+        require(!initialized, "LBPManager: already initialized");
+        require(
+            _beneficiary != address(0),
+            "LBPManager: _beneficiary can not be zero address"
+        );
+
+        initialized = true;
+        admin = msg.sender;
+        primeDaoFeePercentage = _primeDaoFeePercentage;
+        beneficiary = _beneficiary;
+        amounts = _amounts;
+        tokenList = _tokenList;
+
+        lbp = ILBP(
+            ILBPFactory(_LBPFactory).create(
+                _name,
+                _symbol,
+                _tokenList,
+                _weights,
+                _swapFeePercentage,
+                address(this),
+                true // SwapEnabled is set to true at pool creation.
+            )
+        );
+
+        lbp.updateWeightsGradually(
+            _startTimeEndTime[0],
+            _startTimeEndTime[1],
+            _endWeights
+        );
+
+        return address(lbp);
+    }
+
+    /**
+     * @dev                             Subtracts the primeDaoFee and adds liquidity to the LBP.
+     * @param _tokenList                Numerically sorted array (ascending) containing two addresses:
+                                            - The address of the project token being distributed.
+                                            - The address of the funding token being exchanged for the project token.
+     * @param _projectTokenIndex        Index for the _tokenList array for the funding token.
+     * @param _sender                   Address of the liquidity provider.
+     * @param _userData                 UserData object that specifies the type of join.
+     */
+    function addLiquidity(
+        IERC20[] memory _tokenList,
+        uint8 _projectTokenIndex,
```

Comment:
Afaiu, the only way to check that in a bulletproof manner onchain is to compare against the addresses of funding tokens, which we know and which are fixed (?) But maybe it would be sufficient to check that the weight of the alleged project token is smaller than the weight of the alleged funding token? Anyways, I'm curious why you think we need to implement additional safeguards against faulty inputs from the UI here :)

Reply:
So we had a call with Doug on Friday. He said it would be better to sort it onchain because there're only two tokens, which should not be costly but just a direct comparison `>`. This change will be implemented in a different PR. I already started on it but encountered an issue about which I've sent a question to Doug.

Reply:
So we will probably end up with the project and funding token always coming in a certain order, and a quick comparison to sort the array

Reply:
> So we will probably end up with the project and funding token always coming in a certain order, and a quick comparison to sort the array

sounds good to me